gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
I think delimiter issue is sorted in learn_vocab [here](https://github.com/prakhar2b/gensim/blob/c0d17c40ad635d29ce06baa3626266ddd9df33a5/gensim/models/phrases.py#L171). 

Just for information, what could be those dangerous errors ? The idea is simply to do the conversion for incoming token in `__getitem__` to match the encoding in vocab.